.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,27 @@
+name: Go Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  go-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+
+      - run: go version
+
+      - name: Run GARM Go Tests
+        run: make go-test
```

Comment:
Don't forget the makefile. Or you could just call `go test`

Reply:
Added the Makefile now